wolfcrypt/src/rsa.c

```diff
@@ -1609,13 +1609,40 @@ int wc_RsaDirect(byte* in, word32 inLen, byte* out, word32* outSz,
 int wc_RsaFunction(const byte* in, word32 inLen, byte* out,
                           word32* outLen, int type, RsaKey* key, WC_RNG* rng)
 {
-    int ret;
+    int ret = 0;
 
     if (key == NULL || in == NULL || inLen == 0 || out == NULL ||
             outLen == NULL || *outLen == 0 || type == RSA_TYPE_UNKNOWN) {
         return BAD_FUNC_ARG;
     }
 
+    if (type == RSA_PRIVATE_DECRYPT) {
```

Comment:
I'm not sure we need this check multiple times in async mode.  Also, can the check be on by default but with a switch to disable if needed?

Reply:
Sure thing.